lib/identity/index.js

```diff
@@ -145,17 +145,65 @@ export default class Identity extends CryptoConsumer {
       }
     )
     /* eslint-enable */
-    /* eslint-disable-next-line no-warning-comments */
-    // TODO: Write recovery note based on final parameters
+    const brokerClientID = '00000000-0000-0000-0000-000000000000'
+    const brokerInfo = idClient.storageClient.clientInfo(brokerClientID)
+    const brokerKeyNoteName = await this.crypto.genericHash(
+      `broker${username}${this.config.realmId}key`
+    )
+    const brokerKeyBytes = await this.crypto.randomBytes(64)
+    const brokerKey = await this.crypto.b64encode(brokerKeyBytes)
+    const brokerNoteCreds = await deriveNoteCreds(
+      this.config,
+      this.crypto,
+      username,
+      brokerKey,
+      true
+    )
+    /* eslint-disable camelcase */
+    const brokerKeyNote = await idClient.storageClient.writeNote(
+      { brokerKey, username },
+      brokerInfo.publicKey.curve25519,
+      brokerInfo.signing_key.ed25519,
+      {
+        id_string: brokerKeyNoteName,
+        max_views: -1,
+        expires: false,
+        eacp: {
+          email_otp: {
+            email_address: email,
+            template: 'password_recovery',
+            provider_link: this.config.brokerTargetUrl
+          }
+        }
+      }
+    )
+    await idClient.storageClient.writeNote(
+      idClient.serialize(),
+      brokerNoteCreds.cryptoKeyPair.publicKey,
+      brokerNoteCreds.signingKeyPair.publicKey,
+      {
+        id_string: brokerNoteCreds.noteID,
+        max_views: -1,
+        expires: false,
+        eacp: {
+          note_access: {
+            note_id: brokerKeyNote.noteId
+          }
+        }
+      }
+    )
+    /* eslint-enable */
+
     return idClient
   }
 
-  async login(username, password) {
+  async login(username, password, broker = false) {
```

Comment:
Comment explaining that broker=true login is not meant for the typical user might be helpful here, or pulling it out into a separate function. It's not immediately clear why this broker bool is here and this is probably one of the most common functions a user of the sdk will calling.

Reply:
Good point, all of these items need doc comments. I'll see what I can do about adding them.